projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -37,8 +48,14 @@ export default function Edit( props ) {
 	 */
 	const showControls = applyFilters( 'jetpack.RecurringPayments.showControls', true, clientId );
 
+	const availability = getJetpackExtensionAvailability( 'recurring-payments' );
+	const hasWpcomUpgradeNudge =
+		! availability.available && 'missing_plan' === availability.unavailableReason;
+	const showJetpackUpgradeNudge =
+		!! upgradeUrl && ! hasWpcomUpgradeNudge && ! isPremiumContentChild;
+
```

Comment:
This can be simplified to:
`	const showJetpackUpgradeNudge = 'missing_plan' !== availability?.unavailableReason && !! upgradeUrl;
`
The unavailableReason is set only when available is false, so checking that available is false is redundant. On the other hand, we don't need to check if ! isPremiumContentChild since the Premium Content block is only registered for WPCOM and Atomic sites so when we do need to show the Jetpack upgrade nudge we are 100% sure that the Payment Button block isn't in a Premium Conteng block.